src/NLog.StructuredLogging.Json/LoggerExtensions.cs

```diff
@@ -91,11 +93,27 @@ private static void TransferDataToLogEventProperties(LogEventInfo log, object lo
                 return;
             }
 
-            logProperties.GetType()
-                .GetProperties()
-                .Where(p => p.GetIndexParameters().Length == 0)
-                .ToList()
-                .ForEach(x => log.Properties.Add(x.Name, x.GetValue(logProperties)));
+            if (IsDictionary(logProperties))
+            {                
+                foreach (var logProperty in (IDictionary<object, object>) logProperties)
+                {
+                    log.Properties.Add(logProperty.Key, logProperty.Value);
+                };
+            }
+            else
+            {
+                logProperties.GetType()
+                    .GetProperties()
+                    .Where(p => p.GetIndexParameters().Length == 0)
+                    .ToList()
+                    .ForEach(x => log.Properties.Add(x.Name, x.GetValue(logProperties)));
```

Comment:
Can we just change this part too
````
var objectProps =  logProperties.GetType()
   .GetProperties()
   .Where(p => p.GetIndexParameters().Length == 0);

 foreach (var prop in objectProps)
{
   log.Properties.Add(prop .name,  prop.GetValue(logProperties))
}
````

because `.ForEach` is inconsistent and overcomoplex.